src/envoy/core.clj

```diff
@@ -117,10 +134,10 @@
 
 (defn watch-path
   ([path fun]
-   (watch-path path fun {}))
+   (watch-path path fun {:token (System/getenv "CONSUL_TOKEN")}))
```

Comment:
`CONSUL_TOKEN` is an implicit assumption
while a sensible default, in this case, it is best to be explicit since this:
* might be a "`CONSUL_HTTP_TOKEN`"
* or a "`CONSUL_TOKEN`" that points to a different consul
  - this would be fun to debug
* etc.

let's requite token when it is needed
when it is not provided, consul returns a proper error

in this case it is best to communicate the source (consul) behavior vs. assuming